kotlin/src/main/kotlin/com/linecorp/armeria/server/kotlin/auth/AuthService.kt

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.kotlin.auth
+
+import com.linecorp.armeria.common.HttpRequest
+import com.linecorp.armeria.server.HttpService
+import com.linecorp.armeria.server.ServerBuilder
+import com.linecorp.armeria.server.auth.AuthService
+import com.linecorp.armeria.server.auth.AuthServiceBuilder
+import com.linecorp.armeria.server.auth.Authorizer
+
+/**
+ * Decorates this [HttpService] using an [AuthService] with the given [authorizers].
+ */
+fun HttpService.authorizing(vararg authorizers: Authorizer<HttpRequest>): AuthService =
+    decorate(AuthService.newDecorator(*authorizers))
```

Comment:
I don't see this as scalable code. We don't want to update/fix Kotlin DSLs whenever a new decorator is added. 
If you think this gives value to users, we might need a preprocessor that creates shortcut methods by scanning the public Decorators.
We also need to define conventions for the naming of the extension methods.

Reply:
I see, we need to discuss more but this PR is too impatient 😅 
We (include all contributors) would find the best solution with sufficient discussion.